cloud/src/recycler/recycler_operation_log.cpp

```diff
@@ -238,6 +240,22 @@ int OperationLogRecycler::recycle_update_tablet_log(const UpdateTabletLogPB& upd
     return 0;
 }
 
+int OperationLogRecycler::recycle_compaction_log(const CompactionLogPB& compaction_log) {
+    for (const RecycleRowsetPB& recycle_rowset_pb : compaction_log.recycle_rowsets()) {
+        std::string recycle_rowset_value;
+        if (!recycle_rowset_pb.SerializeToString(&recycle_rowset_value)) {
+            LOG_WARNING("failed to serialize RecycleRowsetPB")
+                    .tag("recycle rowset pb", recycle_rowset_pb.ShortDebugString());
+            return -1;
+        }
+        std::string recycle_key =
+                recycle_rowset_key({instance_id_, compaction_log.tablet_id(),
+                                    recycle_rowset_pb.rowset_meta().rowset_id_v2()});
+        kvs_.emplace_back(recycle_key, recycle_rowset_value);
```

Comment:
The input rowset keys, `meta_rowset_load_key` and `meta_rowset_compact_key`, should be removed here.